File: appinventor/components-ios/src/ScaleUnits.swift
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright © 2025 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// https://www.apache.org/licenses/LICENSE-2.0

@objc public class ScaleUnits: NSObject, OptionList {
  @objc public static let Metric = ScaleUnits(1)
  @objc public static let Imperial = ScaleUnits(2)

  private static let LOOKUP: [Int32: ScaleUnits] = generateOptionsLookup(Metric, Imperial)

  let value: Int32

  @objc private init(_ value: Int32) {
    self.value = value
  }

  @objc public func toUnderlyingValue() -> AnyObject {
    return value as AnyObject
  }

  @objc public static func fromUnderlyingValue(_ value: Int32) -> ScaleUnits? {
    return LOOKUP[value]
  }
}

File: appinventor/components/src/com/google/appinventor/components/common/ScaleUnits.java
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright 2020 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

package com.google.appinventor.components.common;

import java.util.HashMap;
import java.util.Map;

/**
 * Defines a ScaleUnits type used by the Map component.
 */
public enum ScaleUnits implements OptionList<Integer> {
  Metric(1),
  Imperial(2);

  private final Integer value;

  ScaleUnits(Integer value) {
    this.value = value;
  }

  public Integer toUnderlyingValue() {
    return value;
  }

  private static final Map<Integer, ScaleUnits> lookup = new HashMap<>();

  static {
    for (ScaleUnits unit : ScaleUnits.values()) {
      lookup.put(unit.toUnderlyingValue(), unit);
    }
  }

  public static ScaleUnits fromUnderlyingValue(Integer unit) {
    return lookup.get(unit);
  }
}
